### PR TITLE
Add isStream

### PR DESCRIPTION
**Pull request by Chalarangelo**
## Description

Inspired by [is-stream](https://www.npmjs.com/package/is-stream). 

**Note:** We could add 3 additional snippets for readable, writable and duplex (read+write) streams.

## PR Type
- [x] Snippets, Tests & Tags (new snippets, updated snippets, re-tagging of snippets, added/updated tests)
- [ ] Scripts & Website & Meta (anything related to files in the [scripts folder](https://github.com/30-seconds/30-seconds-of-code/tree/master/scripts), how the repository's automated procedures work and the website)
- [ ] Glossary & Secondary Features (anything related to the glossary, such as new or updated terms or other secondary features)
- [ ] General, Typos, Misc. & Meta (everything else, typos, general stuff and meta files in the repository - e.g. the issue template)

## Guidelines
- [x] I have read the guidelines in the [CONTRIBUTING](https://github.com/30-seconds/30-seconds-of-code/blob/master/CONTRIBUTING.md) document.

### CODE REVIEW

**Comment by skatcat31** [comment]
I feel like with how confusing streams are and how you can't get them in any other way this is kind of an archive snippet for learning purposes. I really don't know how you'd get a stream you'd have to make sure of since they are so verbose

**Comment by Chalarangelo** [comment]
@skatcat31 Me neither, but apparently that package has seen a ton of usage (1.8m downloads on npm). If the community uses it, we might as well have it in the collection for people who want to understand how to check for streams.

**Comment by skatcat31** [comment]
... We can't add a snippet for `ctrl+f` can we?

**Comment by skatcat31** [comment]
If we're going to move forward with this, should we tag it as a NodeJS one until we can confirm it's working with browser streams(which aren't even supported in FF according to MDN)?

**Comment by Chalarangelo** [comment]
Definitely, that slipped my mind. Thanks for mentioning it!

**Comment by Chalarangelo** [comment]
Alright, that being said I will implement the other 3 (probably tomorrow) and add in this PR, so we can merge the whole set together.

File: snippets/isDuplexStream.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+Check if the value is different from `null`, use `typeof` to check if a value is of type `object` and the `pipe` property is of type `function`.
+Additionally check if the `typeof` the `_read`, `_write` and `_readableState`, `_writableState` properties are `function` and `object` respectively.
+
+```js
+const isDuplexStream = val =>
+  val !== null &&
+  typeof val === 'object' &&
+  typeof val.pipe === 'function' &&
+  typeof val.pipe === 'function' &&
```
[the remark this answers is not in the document]
Comment (Chalarangelo) [reply snippets/isDuplexStream.md]: Fixed in de0cda0.

**Pull request merged**